demisto_sdk/commands/download/downloader.py

```diff
@@ -160,271 +140,312 @@ def __init__(
-                "(either DEMISTO_USERNAME and DEMISTO_PASSWORD, or just DEMISTO_API_KEY) are properly configured.[/red]\n"
-            )
-        logger.info(f"Exception raised when fetching custom content:\nStatus: {e}")
+        if self.download_system_items:
+            if not self.system_item_type:
+                logger.error(
+                    "Error: Missing required parameter for downloading system items: '-it' / '--item-type'."
+                )
+                return False
 
-    def handle_max_retry_error(self, e):
-        logger.info(
-            "\n[red]Verify that the environment variable DEMISTO_BASE_URL is configured properly.[/red]\n"
-        )
-        logger.info(f"Exception raised when fetching custom content:\n{e}")
+            if self.regex:
+                logger.error("Error: RegEx flag ('-r' / '--regex') can only be used for custom content. "
+                             "Use '-i' / '--input' to provide a list of system items to download.")
+                return False
 
-    def download_playbook_yaml(self, playbook_string) -> str:
+            if self.download_all_custom_content:
+                logger.error("Error: All custom content flag ('-a' / '--all') can only be used for custom content. "
+                             "Use '-i' / '--input' to provide a list of system items to download.")
+                return False
+
+        return True
+
+    def filter_custom_content(self, custom_content_objects: dict[str, dict]) -> dict[str, dict]:
         """
-        Downloads the playbook yaml via XSOAR REST API.
-        We should download the file via direct REST API because there are props like scriptName,
-        that playbook from custom content bundle don't contain.
+        Filter custom content data to include only relevant files for the current download command.
+        The function also updates self.input_file with names of content matching the filter.
 
-        If download will fail, then we will return the original playbook_string we received (probably from the bundle)
+        Args:
+            custom_content_objects (dict[str, dict]): A dictionary mapping custom content names
+                to their corresponding objects to filter.
+
+        Returns:
+            dict[str, dict]: A new custom content objects dict with filtered items.
         """
-        existing = yaml.load(playbook_string)
-        playbook_id = existing.get("id")
-        playbook_name = existing.get("name")
-        if (
-            playbook_id
-            and playbook_name
-            and (playbook_name in self.input_files or self.all_custom_content)
-        ):
-            # download the playbook yaml in case playbook name appears in the input_files
-            # or --all-custom-content flag is true
+        file_name_to_content_name_map = {
+            key: value["name"] for key, value in custom_content_objects.items()
+        }
+        filtered_custom_content_objects: dict[str, dict] = {}
```

Comment:
Already discussed this with @ilaner, not really because we don't know what the content types of the files we download are (before parsing them with these dictionaries and running some functions).

Graph models deduce a content item's type by its content path (Integration, Playbook, etc.) which isn't relevant here (all are downloaded together in a tar package with no directory)

Reply:
We can also find the content type by its content